manager/src/main/java/com/redhat/service/bridge/manager/connectors/creation/CheckConnectorAvailableWorker.java

```diff
@@ -0,0 +1,69 @@
+package com.redhat.service.bridge.manager.connectors.creation;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Objects;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.openshift.cloud.api.connector.models.Connector;
+import com.redhat.service.bridge.infra.exceptions.BridgeErrorDAO;
+import com.redhat.service.bridge.infra.exceptions.definitions.platform.ConnectorNotFoundException;
+import com.redhat.service.bridge.infra.models.dto.ConnectorStatus;
+import com.redhat.service.bridge.manager.connectors.AbstractConnectorWorker;
+import com.redhat.service.bridge.manager.connectors.ConnectorsApiClient;
+import com.redhat.service.bridge.manager.connectors.Events;
+import com.redhat.service.bridge.manager.models.ConnectorEntity;
+
+import io.quarkus.vertx.ConsumeEvent;
+
+@ApplicationScoped
+public class CheckConnectorAvailableWorker extends AbstractConnectorWorker<Connector> {
+
+    @Inject
+    ConnectorsApiClient connectorsApiClient;
+
+    @Inject
+    BridgeErrorDAO bridgeErrors;
+
+    @ConsumeEvent(value = Events.CONNECTOR_MANAGED_CONNECTOR_CREATED_EVENT, blocking = true)
+    public void consume(ConnectorEntity connectorEntity) {
+        execute(connectorEntity);
+    }
+
+    @Override
+    protected Connector callExternalService(ConnectorEntity connectorEntity) {
+        Connector connector = connectorsApiClient.getConnector(connectorEntity);
+        if (Objects.isNull(connector)) {
+            throw new ConnectorNotFoundException(bridgeErrors.findByException(ConnectorNotFoundException.class).getReason());
+        }
+        return connector;
+    }
+
+    @Override
+    protected ConnectorEntity updateEntityForSuccess(ConnectorEntity connectorEntity, Connector connector) {
+        connectorEntity.setPublishedAt(ZonedDateTime.now(ZoneOffset.UTC));
+        connectorEntity.setStatus(ConnectorStatus.READY);
+        return connectorEntity;
+    }
+
+    @Override
+    public void errorWhileCalling(Exception error, ConnectorEntity connectorEntity) {
+        super.errorWhileCalling(error, connectorEntity);
+
+        // If the Connector could not be found propagate the exception so VertX can re-try for us.
```

Comment:
I didn't know one could use Vert.x for retrying, it's an interesting pattern. 

I was expecting some kind of for loop / polling inside the `callExternalService` method instead, which could be more explicit

Reply:
Well.. that'll teach me to try and be smart. 

Writing a Unit Test revealed this approach didn't work.. throwing the exception causes the TX to be rolled back which lead to other problems relating to state. So I've added a new commit that refactors the re-try mechanism to `ConnectorsOrchestrator` (as is the case for other re-tries when steps fail). I've now scheduled `ConnectorsOrchestrator` too.